Automattic/jetpack
# Add review prompt for social plugin

#### Changes proposed in this Pull Request:

* This PR adds a review request prompt for the Jetpack Social plugin after a post is shared to connected social media accounts

#### Other information:

#### Jetpack product discussion


pbNhbs-3wm-p2

#### Does this pull request change what data or activity we track or use?


Yes, this PR adds 3 new analytics events around the new review prompt:
- `jetpack_social_plugin_review_prompt_view` (recorded when review prompt is rendered)
- `jetpack_social_plugin_review_prompt_new_review_click` (recorded when "Leave a review" is clicked)
- `jetpack_social_plugin_review_prompt_dismiss_click` (recorded when "Dismissed" is clicked)

#### Testing instructions:






- You will need a Jetpack site with the Jetpack Social plugin installed
- Note that if you also have the Jetpack plugin installed, you will need to deactivate it for this test
- From your test site, you can apply this branch to Jetpack Social using the Jetpack Beta testing plugin
- Connect at least one social account for your site using Jetpack Social
- After your social account is connected, create a new post and publish it. Social sharing should be on by default and your post should be shared to your connected social media account
- In the post-publish sidebar of the editor, you should see the new review prompt

![Screen Shot 2022-12-13 at 11 41 10 AM](https://user-images.githubusercontent.com/18016357/207461044-e3049bf5-1017-41c0-a121-38c2a8f9405d.png)

- When the prompt renders, the `jetpack_social_plugin_review_prompt_view` analytics event should fire. Search for this in your browser's network panel to confirm.
- Confirm that clicking on "Leave a Review" takes you to the reviews page for the Jetpack Social plugin
- Clicking on "Leave a Review" should also fire the `jetpack_social_plugin_review_prompt_new_review_click` analytics event, search for this in your browser's network panel.
- Do not dismiss the prompt yet, we'll test that later :)
- Now, go create another new post. This time, use the Jetpack social sidebar (click on the megaphone icon in the upper right hand corner of the editor) to disable social sharing for this post
- Publish this new post with social sharing disabled. The review prompt should not show in the post-publish sidebar since sharing was disabled.
- Run this test again with a 3rd post, but disable the individual sharing account(s) before publishing instead of using the toggle to disable social sharing all together. You should not see the review prompt when publishing this 3rd post.
- Publish a 4th post with social sharing on, you should see the review prompt again. Click "dismiss" to dismiss the prompt.
- Clicking "dismiss" should fire the `jetpack_social_plugin_review_prompt_dismiss_click` analytics event. Search in your browser's network console to confirm.
- When dismissed, the prompt should disappear immediately and will not show again for future shared posts.
- If you need to get the review prompt to show again for further testing, you can set the option `jetpack-social_review_prompt_dismissed` to `0` using the `wp-cli`
